src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -282,12 +284,23 @@ private AdjustmentRule GetAdjustmentRuleForAmbiguousOffsets(DateTime adjustedTim
         /// If the specified rule is the first AdjustmentRule, or it isn't in _adjustmentRules,
         /// then the specified rule is returned.
         /// </summary>
-        private AdjustmentRule GetPreviousAdjustmentRule(AdjustmentRule rule)
+        private AdjustmentRule GetPreviousAdjustmentRule(AdjustmentRule rule, int? ruleIndex)
```

Comment:
>GetPreviousAdjustmentRule(AdjustmentRule rule, int? ruleIndex) [](start = 31, length = 62)

Is it worth to have overload of this method which takes initialized index? this can eliminate the checks of the index initialization